src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -341,20 +385,53 @@ func (r *ClusterReconciler) createBootstrapStatefulSet(
 								},
 							},
 						},
+						{
+							Name:	"configmap-dir",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: configMapName,
+									},
+									DefaultMode:	&configMapDefaultMode,
+								},
+							},
+						},
+						{
+							Name:	"config-dir",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					InitContainers: []corev1.Container{
+						{
+							Name:		"redpanda-configurator",
+							Image:		cluster.Spec.Image + ":" + cluster.Spec.Version,
+							Command:	[]string{"/bin/sh", "-c"},
+							Args:		[]string{configuratorPath},
```

Comment:
although configuration sounds OK as a first-pass to this architecture. It might be worth (in a separate series) to actually integrate natively w/ RPK RPC & binary itself. It's written in Go. So instead of issuing shell-to-rpk we can just invoke the API itself no?

I think configurator could be ok until we have stability of what we want the configurator to do.

Reply:
Let's include this in our discussion.

We use `rpk` in the container command because the configuration is specific to the container, and not common across the StatefulSet. We could eventualy build a configurator binary that integrates natively with RPK instead of shell commands, if that's what you meant.